fdbclient/include/fdbclient/StorageServerInterface.h

```diff
@@ -1266,6 +1270,51 @@ struct GetHotShardsRequest {
 	}
 };
 
+struct CheckSumMetaData {
+	constexpr static FileIdentifier file_identifier = 3828142;
+	KeyRange range;
+	Version version;
+	StringRef checkSumValue;
+
+	CheckSumMetaData() {}
+	explicit CheckSumMetaData(KeyRange range, Version version, StringRef checkSumValue)
+	  : range(range), version(version), checkSumValue(checkSumValue) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, range, version, checkSumValue);
+	}
+};
+
+struct GetStorageCheckSumReply {
+	constexpr static FileIdentifier file_identifier = 3828143;
+	std::vector<CheckSumMetaData> checkSum;
+
+	GetStorageCheckSumReply() {}
+	explicit GetStorageCheckSumReply(std::vector<CheckSumMetaData> checkSum) : checkSum(checkSum) {}
```

Comment:
nit: here checkSum is copied twice,

```
const std::vector<checksumMetadata>& checksum
```

should avoid one copy while another possible consideration is

```
(std::vector<CheckSumMetaData>&& checkSum_) : checkSum(std::move(checkSum_))